src/black/__init__.py

```diff
@@ -660,14 +662,14 @@ def get_sources(
         elif p.is_dir():
             if exclude is None:
                 exclude = re_compile_maybe_verbose(DEFAULT_EXCLUDES)
-                gitignore = get_gitignore(root)
+                root_gitignore = get_gitignore(root)
                 p_gitignore = get_gitignore(p)
                 # No need to use p's gitignore if it is identical to root's gitignore
                 # (i.e. root and p point to the same directory).
```

Comment:
What if there are more than two levels? For example, we're formatting `a/b/c/` and `a/.gitignore/`, `a/b/.gitignore`, and `a/b/c/.gitignore` all exist.

Reply:
Good question. Right now, if the command is `black a/b/c/` then `a/b/.gitignore` is completely skipped while `a/.gitignore` and `a/b/c/.gitignore` are captured by `root_gitignore` and `p_gitignore`, respectively. What's the expected behavior?

Reply:
Should this be a separate issue/PR? If its an undefined behavior, then it should be documented properly I think

Reply:
> What if there are more than two levels? For example, we're formatting `a/b/c/` and `a/.gitignore/`, `a/b/.gitignore`, and `a/b/c/.gitignore` all exist.

The current behavior seems to be that `black a/b/c` only detects `a/b/c/.gitignore` and completely ignores other gitignore files. This PR implements the same behavior